streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamFlatMapTest.java

```diff
@@ -86,4 +88,12 @@ public void testFlatMap() {
             assertEquals(expected[i], supplier.theCapturedProcessor().processed().get(i));
         }
     }
+
+    @Test
+    public void testKeyValueMapperResultNotNull() {
+        final KStreamFlatMap<String, Integer, String, Integer> supplier = new KStreamFlatMap<>((key, value) -> null);
+        final Record<String, Integer> record = new Record<>("K", 0, 0L);
+        assertThrows(String.format("KeyValueMapper can't return null from mapping the record: %s", record),
+                NullPointerException.class, () -> supplier.get().process(record));
```

Comment:
Just copying over the suggestion to here, so it's easy to find

```suggestion
       final Throwable throwable = assertThrows(NullPointerException.class, () -> supplier.get().process(record));
       assertEquals(throwable.getMessage(), String.format("KeyValueMapper can't return null from mapping the record: %s", record));
```